package/MDAnalysis/analysis/density.py

```diff
@@ -180,11 +177,259 @@
+            # Issue 2372: padding is ignored, defaults to 2.0 therefore warn
+            if self._padding > 0:
+                msg = ("Box padding (currently set at {0}) "
+                       "is not used in user defined grids.".format(self._padding))
+                warnings.warn(msg)
+                logger.warning(msg)
+            # Generate a copy of smin/smax from coords to later check if the
+            # defined box might be too small for the selection
+            smin = np.min(coord, axis=0)
+            smax = np.max(coord, axis=0)
+            # Overwrite smin/smax with user defined values
+            smin, smax = _set_user_grid(self._gridcenter, self._xdim,
+                                        self._ydim, self._zdim, smin, smax)
+        else:
+            # Make the box bigger to avoid as much as possible 'outlier'. This
+            # is important if the sites are defined at a high density: in this
+            # case the bulk regions don't have to be close to 1 * n0 but can
+            # be less. It's much more difficult to deal with outliers.  The
+            # ideal solution would use images: implement 'looking across the
+            # periodic boundaries' but that gets complicated when the box
+            # rotates due to RMS fitting.
+            smin = np.min(coord, axis=0) - self._padding
+            smax = np.max(coord, axis=0) + self._padding
+        BINS = fixedwidth_bins(self._delta, smin, smax)
+        arange = np.transpose(np.vstack((BINS['min'], BINS['max'])))
+        bins = BINS['Nbins']
+        # create empty grid with the right dimensions (and get the edges)
+        grid, edges = np.histogramdd(np.zeros((1, 3)), bins=bins,
+                                     range=arange, normed=False)
+        grid *= 0.0
+        self._grid = grid
+        self._edges = edges
+        self._arange = arange
+        self._bins = bins
+        #: density as a :class:`Density`
+        self.density = None
+
+    def _single_frame(self):
+        h, _ = np.histogramdd(self._atomgroup.positions,
+                              bins=self._bins, range=self._arange,
+                              normed=False)
+        # manually reduce (not yet part of AnalysisBase but exists in pmda.parallel.AnalysisBase)
+        self._reduce(self._grid, h)
+
+    def _conclude(self):
+        # average:
+        self._grid /= float(self.n_frames)
+
+        metadata = self._metadata if self._metadata is not None else {}
+        metadata['topology'] = metadata['psf'] = self._atomgroup.universe.filename
+        metadata['trajectory'] = metadata['dcd'] = self._trajectory.filename
+        metadata['n_frames'] = self.n_frames
+        metadata['totaltime'] = self._trajectory.totaltime
+        metadata['dt'] = self._trajectory.dt
+        metadata['time_unit'] = "ps"
+        parameters = self._parameters if self._parameters is not None else {}
+        parameters['isDensity'] = False  # must override
```

Comment:
`isDensity` tells `Density` if it is a density (something per volume) or just counts. IIRC the idea had been that `parameters` could hold more things but obviously that never happened so it would be better to just have a `isDensity` kwarg.

`isDensity` is necessary so that one can initialize `Density` from either a histogram or a real density. In this case we know that we have a histogram so no matter what a user might have set in `parameters`, we must override it ... nevermind that there's only one thing.... so that's why I am removing it :-).

Reply:
Changing `parameters` in `Density` is a good idea... 2.0